File: card-games/card-games-core/src/main/c++/StandardPlayingCardDeck.cpp
#include "StandardPlayingCardDeck.h"

#include <iostream>

StandardPlayingCardDeck::StandardPlayingCardDeck() {
  for(int i = StandardPlayingCardSuit::HEART; i != StandardPlayingCardSuit::CLUB; i++) {
    StandardPlayingCard card(static_cast<StandardPlayingCardSuit>(i));
    this->push_back(card);
    cout << card << endl;
  }
}

File: card-games/card-games-core/src/main/java/net/cdg29/games/card/StandardPlayingCardDeck.java
package net.cdg29.games.card;

public class StandardPlayingCardDeck extends CardPile<StandardPlayingCard> {
  public StandardPlayingCardDeck() {
    super();
    for(StandardPlayingCardSuit suit : StandardPlayingCardSuit.values()) {
      for(StandardPlayingCardRank rank : StandardPlayingCardRank.values()) {
        add(new StandardPlayingCard(suit, rank));
      }
    }
  }
}
